### PR TITLE
Style completed states

### PR DESCRIPTION
**Pull request by brabbuss**
### What does this pull request do?
- Add link to a task within `Task` itself, not in `DailyMission` view
- add completed state with logic and styling to `Task` component
- The task will be updated in state and in the DB 
- Completed task cannot be clicked through - however there is no logic to stop manually navigating to the correct endpoint (though this is a MCP++ sort of error handling in my opinion)

### Next steps
- will need to ensure that every task.category has something to render and renders the correct task component

### CODE REVIEW

File: src/modules/task/Task.js [hunk 1]
```diff
@@ -38,13 +37,25 @@ const useStyles = makeStyles((theme) => ({
...
       color: "gold",
       margin: 0,
       marginRight: ".4em",
     },
   },
+  pulse: {
+    animation: "$pulse infinite alternate .75s ease",
+    color: "lightgreen !important",
+  },
+  "@keyframes pulse": {
```
Comment (baileydunning) [inline src/modules/task/Task.js:51]: ooooh yeah, gotta love a good pulse effect

File: src/modules/task/Task.js [hunk 2]
```diff
@@ -88,31 +110,89 @@ const useStyles = makeStyles((theme) => ({
...
+  const classes = useStyles(props);
   const {
-    attributes: { name, description, category, points },
+    attributes: {
+      mission_id,
+      task_name,
+      task_description,
+      task_category,
+      points,
+      photoIsRequired,
```
Comment (baileydunning) [inline src/modules/task/Task.js:130]: this is just called `photo` on the BE

**Review by baileydunning — APPROVED** [review]

File: src/modules/task/Task.js (hunk 1 above)
Comment (lo-la-do-li) [reply src/modules/task/Task.js:51]: Love that!

File: src/modules/task/Task.js [hunk 3]
```diff
@@ -88,31 +110,89 @@ const useStyles = makeStyles((theme) => ({
...
+      return (
+        <span className={classes.taskText}>
+          <h1 style={{ marginRight: ".5em" }}>Agent Task</h1>
+          <p>{task_name}</p>
+        </span>
+      );
+    }
+  };
+
+  const renderTaskCard = () => {
```
Comment (lo-la-do-li) [inline src/modules/task/Task.js:155]: Nice job setting up the completed conditional! Woohoo!

**Review by lo-la-do-li — APPROVED** [review]

**Pull request merged**